pkg/process/metadata/parser/java/spring.go

```diff
@@ -0,0 +1,205 @@
+								log.Trace("Loaded", f.Name)
+								val, ok := ret[profile]
+								if !ok {
+									val = &props.Combined{Sources: []props.PropertyGetter{}}
+									ret[profile] = val
+								}
+								val.Sources = append(val.Sources, *source)
+							}
+						}
+						break
+					}
+				}
+			}
+		}
+	}
+	return ret
+}
+
+// GetSpringBootAppName tries to autodetect the name of a spring boot application given its working dir,
+// the jar path and the application arguments
+func GetSpringBootAppName(cwd string, jarname string, args []string) (string, error) {
+	absName := abs(jarname, cwd)
+	log.Tracef("open jar archive %s", absName)
+	archive, err := zip.OpenReader(jarname)
+	if err != nil {
+		return "", err
+	}
+	defer archive.Close()
+	ok, err := IsSpringBootArchive(&archive.Reader)
+	if !ok || err != nil {
+		log.Trace("Not a spring boot archive.")
+		return "", nil
+	}
+
+	combined := &props.Combined{Sources: []props.PropertyGetter{
+		newArgumentSource(args, "--"),
+		newArgumentSource(args, "-D"),
+		// TODO: add the env variable of the process being introspected
+		// I did not found in the agent packages something to do it cross os.
+		// On linux one trivial way is to look in the proc fs
+		//&props.Environment{Normalize: true},
+	}}
+
+	// resolved properties referring to other properties just in case...
+	conf := &props.Configuration{Props: props.NewExpander(combined)}
+	// Looking in the environment (sysprops, arguments) first
+	appname, ok := conf.Get(appnamePropName)
+	if !ok {
+		locations := strings.Split(combined.GetDefault(locationPropName, defaultLocations), ";")
+		confname := combined.GetDefault(configPropName, defaultConfigName)
+		var profiles []string
+		rawProfile, ok := combined.Get(activeProfilesPropName)
+		if ok && len(rawProfile) > 0 {
+			profiles = strings.Split(rawProfile, ",")
+		}
+		log.Trace("Parsing spring configuration from locations", locations, "with configuration name", confname)
+		files, classpaths := parseURI(locations, confname, profiles, cwd)
+		log.Trace("Loading filesystem properties from", files)
+		fileSources := scanSourcesFromFileSystem(files)
+		log.Trace("Loading archive properties from classpath", classpaths)
+		classpathSources := newSpringBootArchiveSourceFromReader(&archive.Reader, classpaths)
+		//assemble by profile
+		for _, profile := range append(profiles, "") {
+			val, ok := fileSources[profile]
+			if ok {
+				combined.Sources = append(combined.Sources, val)
+			}
+			val, ok = classpathSources[profile]
+			if ok {
+				combined.Sources = append(combined.Sources, val)
+			}
+		}
+
+		if err != nil {
+			return "", err
+		}
+
+		appname, _ = conf.Get(appnamePropName)
+	}
+	return appname, nil
+}
+
+// IsSpringBootArchive heuristically determines if a jar archive is a spring boot packaged jar
+func IsSpringBootArchive(reader *zip.Reader) (bool, error) {
+	for _, f := range reader.File {
+		if f.Name == "BOOT-INF/" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
```

Comment:
you never return error

```suggestion
// IsSpringBootArchive heuristically determines if a jar archive is a spring boot packaged jar
func IsSpringBootArchive(reader *zip.Reader) bool {
	for _, f := range reader.File {
		if f.Name == "BOOT-INF/" {
			return true
		}
	}
	return false
}
```

Reply:
that's true